### PR TITLE
Allow admins to remove entries from some config arrays

### PR DESCRIPTION
**Pull request by neumantm**
This is done by setting the respective value to null

Resolves: #1154

### CODE REVIEW

**Comment by neumantm** [comment]
Tried and confirmed working for `themes`, `tshirt_sizes`, and `footer_items`.
Should also work for the others (but `header_items` currently has no entry to disable in the default config)

File: src/Config/ConfigServiceProvider.php [hunk 1]
```diff
@@ -39,6 +40,12 @@ public function register(): void
             $config->set($configuration);
         }
 
+        foreach ($this->configVarsToPruneNulls as $key) {
+            $config->set($key, array_filter($config->get($key), function($v) {
```
Comment (MyIgel) [inline src/Config/ConfigServiceProvider.php:44]: Please run the style checks documented in `DEVELOPMENT.md`:
```
FILE: /var/www/src/Config/ConfigServiceProvider.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 44 | ERROR | [x] Expected 1 space after FUNCTION keyword; 0 found
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

File: src/Config/ConfigServiceProvider.php [hunk 2]
```diff
@@ -13,6 +13,7 @@
 class ConfigServiceProvider extends ServiceProvider
 {
     protected array $configFiles = ['app.php', 'config.default.php', 'config.php'];
+    protected array $configVarsToPruneNulls = ['themes', 'tshirt_sizes', 'headers', 'header_items', 'footer_items'];
```
Comment (MyIgel) [inline src/Config/ConfigServiceProvider.php:16]: Eventually also `locales`?

**Review by MyIgel — CHANGES_REQUESTED** [review]
Hi, thank you for that PR!
There are some smaller things to do here, see the following comments.
In addition it would be nice to have a comment on the corresponding options stating that they can be null'd & the readme should also be updated accordingly.

File: src/Config/ConfigServiceProvider.php [hunk 3]
```diff
@@ -39,6 +40,12 @@ public function register(): void
             $config->set($configuration);
         }
 
+        foreach ($this->configVarsToPruneNulls as $key) {
+            $config->set($key, array_filter($config->get($key), function($v) {
+                return $v !== null;
```
Comment (MyIgel) [inline src/Config/ConfigServiceProvider.php:45]:
```suggestion
                return !is_null($v);
```

**Comment by MyIgel** [comment]
Hi @neumantm, nay news here?

**Comment by neumantm** [comment]
> Hi @neumantm, nay news here?

Sorry, did not get to it until now.
Added a commit, please re-review.

**Comment by neumantm** [comment]
>  Added a commit, please re-review.

@MyIgel ^

File: tests/Unit/Config/ConfigServiceProviderTest.php [hunk 4]
```diff
@@ -19,6 +19,9 @@ class ConfigServiceProviderTest extends ServiceProviderTest
 {
     use ArraySubsetAsserts;
 
+    private array $configVarsWhereNullIsPruned = ['themes', 'tshirt_sizes', 'headers',
+                                                  'header_items', 'footer_items', 'locales'];
```
Comment (MyIgel) [inline tests/Unit/Config/ConfigServiceProviderTest.php:23]: We have still linter errors in this file
```
tests/Unit/Config/ConfigServiceProviderTest.php:
	23: Wrong amount of left-padding spaces(want multiple of 4)
	36: Wrong amount of left-padding spaces(want multiple of 4)
	37: Wrong amount of left-padding spaces(want multiple of 4)
	39: Wrong amount of left-padding spaces(want multiple of 4)
	40: Wrong amount of left-padding spaces(want multiple of 4)
	48: Wrong amount of left-padding spaces(want multiple of 4)
	51: Wrong amount of left-padding spaces(want multiple of 4)
	52: Wrong amount of left-padding spaces(want multiple of 4)
	53: Wrong amount of left-padding spaces(want multiple of 4)
	54: Wrong amount of left-padding spaces(want multiple of 4)
	66: Wrong amount of left-padding spaces(want multiple of 4)
```

File: src/Config/ConfigServiceProvider.php [hunk 5]
```diff
@@ -14,6 +14,10 @@ class ConfigServiceProvider extends ServiceProvider
 {
     protected array $configFiles = ['app.php', 'config.default.php', 'config.php'];
 
+    # remember to update ConfigServiceProviderTest, config.default.php, and README.md
+    protected array $configVarsToPruneNulls = ['themes', 'tshirt_sizes', 'headers',
+                                               'header_items', 'footer_items', 'locales'];
```
Comment (MyIgel) [inline src/Config/ConfigServiceProvider.php:19]: There are still linter errors:
```
src/Config/ConfigServiceProvider.php:
	19: Wrong amount of left-padding spaces(want multiple of 4)
```

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: tests/Unit/Config/ConfigServiceProviderTest.php (hunk 4 above)
Comment (neumantm) [reply tests/Unit/Config/ConfigServiceProviderTest.php:23]: Weird, `docker exec engelsystem_dev-es_workspace-1 composer run phpcs` (as suggested in DEVELOPMENT.md) doesn't report these errors.
Will fix.

**Pull request merged**